src/ray/common/scheduling/scheduling_ids.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+#define RAY_COMMON_SCHEDULING_SCHEDULING_IDS_H
+
+#include <iostream>
+#include <map>
+#include <unordered_map>
+#include <string>
+#include <iterator>
+#include <stdlib.h>
+#include <chrono>
+#include <vector>
+using namespace std;
+
+/// Class to map string IDs to unique integer IDs and back.
+class ScheduleIds {
+  unordered_map<string, int64_t> string_to_int;
```

Comment:
Maybe use more specific terminology here? `resource_label_to_id_` would be one idea. Note that private fields end with underscores in our convention.